extendr-api/src/wrapper/externalptr.rs

```diff
@@ -260,3 +285,73 @@ impl<T: Debug> std::fmt::Debug for ExternalPtr<T> {
         (&**self as &T).fmt(f)
     }
 }
+
+impl<T> AsRef<T> for ExternalPtr<T> {
+    fn as_ref(&self) -> &T {
+        self.addr()
+    }
+}
+
+impl<T> AsMut<T> for ExternalPtr<T> {
+    fn as_mut(&mut self) -> &mut T {
+        self.addr_mut()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use extendr_engine::with_r;
+
+    #[test]
+    fn partial_eq_of_externalptr() {
```

Comment:
I'd prefer to have this split into a couple more specific tests. Or at least into 'these should be equal', 'these should be unequal', etc. Then if one test fails, it will be easier to understand what is going on.

Reply:
I've tried to split it. Can you look at it again plase?